### PR TITLE
feature: S3UTILS-60 followerDiff tool

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Introduce a tool "followerDiff" in the CompareRaftMembers directory to
compare a follower's metadata databases against the leader's view

### CODE REVIEW

File: CompareRaftMembers/followerDiff.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    const db = new Level(dbPath);
+    const dbRawStream = db.createReadStream();
+    const dbListStream = new DBListStream({ dbName: path.basename(dbPath) });
+    const diffStream = new DiffStream({
+        bucketdHost: BUCKETD_HOST,
+        bucketdPort: BUCKETD_PORT,
+        digestsDb,
+    });
+
+    dbRawStream.pipe(dbListStream).pipe(diffStream);
```
Comment (watsaqat) [inline CompareRaftMembers/followerDiff.js:134]: minor readability suggestion:

`
dbRawStream
    .pipe(dbListStream)
    .pipe(diffStream);
`

Comment (watsaqat) [inline CompareRaftMembers/followerDiff.js]: minor readability suggestion:

```
dbRawStream
    .pipe(dbListStream)
    .pipe(diffStream);

File: CompareRaftMembers/followerDiff.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    BUCKETD_HOSTPORT: ip:port of bucketd endpoint
+    DATABASES: space-separated list of databases to scan
+    DIFF_OUTPUT_FILE: file path where diff output will be stored
+
+Optional environment variables:
+    LISTING_DIGESTS_INPUT_DIR: read listing digests from the specified LevelDB database
+    PARALLEL_SCANS: number of databases to scan in parallel (default ${DEFAULT_PARALLEL_SCANS})
+`;
+
+if (!BUCKETD_HOSTPORT) {
```
Comment (tmacro) [inline CompareRaftMembers/followerDiff.js:77]: Might also want to do some small check to make sure its in ip:port format. Maybe just checking if it contains a  `:`

**Review by tmacro — APPROVED** [review]
one small suggestion but otherwise lgtm

**Review by watsaqat — APPROVED** [review]

**Pull request merged**